lib/liquid/variable.rb

```diff
@@ -94,17 +94,15 @@ def render(context)
     private
 
     def parse_filter_expressions(filter_name, unparsed_args)
-      filter_args = []
-      keyword_args = {}
+      result = [filter_name, []]
       unparsed_args.each do |a|
         if matches = a.match(JustTagAttributes)
-          keyword_args[matches[1]] = Expression.parse(matches[2])
+          result << {} unless result[2]
+          result[2][matches[1]] = Expression.parse(matches[2])
         else
-          filter_args << Expression.parse(a)
+          result[1] << Expression.parse(a)
         end
       end
-      result = [filter_name, filter_args]
-      result << keyword_args unless keyword_args.empty?
       result
```

Comment:
I'd find this a bit easier to read with locals for `filter_args` and `keyword_args` and the last line allocate the array of `[filter_name, filter_args, keyword_args]`. It's fine to have the nil 3rd element, performance would be nearly the same or slightly better ([no memory cost on MRI since the array is 3 elements or less](https://gist.github.com/pushrax/bd2d3bb99c46ed0570cfbb7d5e82d742#file-memalloc-rb-L21-L31), and locals are faster to access than array members). The point is just readability though.